cloudinfo/containers.go

```diff
@@ -47,3 +49,68 @@ func (infoSvc *CloudInfoService) GetClusterConfigPath(clusterId string, basePath
 
 	return configDetails.FilePath, nil
 }
+
+// GetAlbState retrieves the details of State corresponding to ALB ID
+// albId: the ID of the ALB
+// Returns the State for an ALB ID.
+func (infoSvc *CloudInfoService) GetAlbState(albId string) (status string, err error) {
+	albConfig, detailedResponse, err := infoSvc.albService.GetClusterALB(infoSvc.albService.NewGetClusterALBOptions(albId))
+	if err != nil {
+		log.Println("Failed to get Cluster ALB details for ", albId, ":", err, "Full Response:", detailedResponse)
+		return "", err
+	}
+
+	// If any specific operation to perform for a state(healthy, critical, pending) is requried.
+	/*	if *albConfig.State == "healthy" {
+		} else if *albConfig.State == "critical" {
+		} else {
+		}
+	*/
+	return *albConfig.State, nil
+}
+
+// GetAlbIds retrieves the list of all ALBs in a cluster
+// clusterId: the ID or name of the cluster
+// Returns a list all ALB IDs in a cluster. If no ALB IDs are returned, then the cluster does not have a portable subnet.
+func (infoSvc *CloudInfoService) GetAlbIds(clusterId string) (ids []string, err error) {
+	clusterAlbs, detailedResponse, err := infoSvc.albService.GetClusterALBs(infoSvc.albService.NewGetClusterALBsOptions(clusterId))
+	if err != nil {
+		log.Println("Failed to get ALB IDs for ", clusterId, ":", err, "Full Response:", detailedResponse)
+		return []string{}, err
+	}
+
+	// don't bother looping if empty
+	if len(clusterAlbs) == 0 {
+		return
+	}
+
+	// loop through clusterAlbs to get albIds
+	for _, clusterAlb := range clusterAlbs {
+		ids = append(ids, *clusterAlb.ID)
+	}
+	return ids, nil
+}
+
+// GetIngressState retrieves the state of each albIds in a cluster
+// clusterId: the ID or name of the cluster
+// Returns a map with albId as key and corresponding state as value
+func (infoSvc *CloudInfoService) GetIngressState(clusterId string) (state map[string]string, err error) {
+	albIds, err := infoSvc.GetAlbIds(clusterId)
+	if err != nil {
+		return state, fmt.Errorf("failed to get ALB IDS for cluster: %v", clusterId)
```

Comment:
return empty string (probably?) instead of "state", and your Errorf is great, except add a "%w" also which will wrap the error.

Reply:
I have initialised a new map with the name, "state" before the if condition.

Returning an empty string will give an error as the return statement expects a map. So, returning the "state" variable as it will be an empty map before the for loop on line 108.